File: go/services/echo_service.go
package services

import (
	"test-service/spec/echo"
	"test-service/spec/models"
)

type EchoService struct{}

func (service *EchoService) EchoBody(body *models.Message) (*models.Message, error) {
	return body, nil
}
func (service *EchoService) EchoQuery(intQuery int, stringQuery string) (*models.Message, error) {
	return &models.Message{IntField: intQuery, StringField: stringQuery}, nil
}
func (service *EchoService) EchoHeader(intHeader int, stringHeader string) (*models.Message, error) {
	return &models.Message{IntField: intHeader, StringField: stringHeader}, nil
}
func (service *EchoService) EchoUrlParams(intUrl int, stringUrl string) (*models.Message, error) {
	return &models.Message{IntField: intUrl, StringField: stringUrl}, nil
}

func (service *EchoService) SameOperationName() (*echo.SameOperationNameResponse, error) {
	return &echo.SameOperationNameResponse{Ok: &echo.Empty}, nil
}

File: java-spring/src/main/java/test_service/services_implementations/EchoService.java
package test_service.services_implementations;

import java.math.BigDecimal;
import java.time.*;
import java.util.*;

import org.springframework.stereotype.Service;

import test_service.models.*;
import test_service.services.echo.*;

@Service("EchoService")
public class EchoService implements IEchoService {
	@Override
	public Message echoBody(Message body) {
		return body;
	}
	@Override
	public Message echoQuery(int intQuery, String stringQuery) {
		return new Message(intQuery, stringQuery);
	}
	@Override
	public Message echoHeader(int intHeader, String stringHeader) {
		return new Message(intHeader, stringHeader);
	}
	@Override
	public Message echoUrlParams(int intUrl, String stringUrl) {
		return new Message(intUrl, stringUrl);
	}
	@Override
	public SameOperationNameResponse sameOperationName() {
		return new SameOperationNameResponseOk();
	}
}
